### PR TITLE
Fix wrong docs on Job.join

### LINKED ISSUE #2615: Job join method documentation
I have found very strange documentation for join method:

`In particular, it means that a parent coroutine invoking join on a child coroutine that was started using launch(coroutineContext) { ... } builder throws CancellationException if the child had crashed, unless a non-standard CoroutineExceptionHandler is installed in the context.`

I'm not sure that CoroutineExceptionHandler will have effect for CancellationException.
Example:

```
fun main() = runBlocking {
    val handler = CoroutineExceptionHandler { _, exception ->
        println("CoroutineExceptionHandler got $exception")
    }
    val job = GlobalScope.launch(handler) {
        val inner = launch { // all this stack of coroutines will get cancelled
            throw IOException() // the original exception
        }
        try {
            inner.join()
        } catch (e: CancellationException) {
            println("handle join")
        }
    }
    job.join()
}
```

Output:
`handle join
CoroutineExceptionHandler got java.io.IOException`

So basically CancellationException will still be thrown regardless any installed handlers.

### PR DESCRIPTION
**Pull request by elizarov**
Fixes #2615

### CODE REVIEW

**Comment by serddmit** [comment]
@elizarov, Don't forget to change documentation for cancelAndJoin, it has the same statement.
https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/cancel-and-join.html

Thanks

**Comment by serddmit** [comment]
Don't forget to change documentation for cancelAndJoin, it has the same statement.
https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/cancel-and-join.html

File: kotlinx-coroutines-core/common/src/Job.kt [hunk 1]
```diff
@@ -252,9 +252,9 @@ public interface Job : CoroutineContext.Element {
      * suspending function is invoked or while it is suspended, this function
      * throws [CancellationException].
      *
-     * In particular, it means that a parent coroutine invoking `join` on a child coroutine that was started using
-     * `launch(coroutineContext) { ... }` builder throws [CancellationException] if the child
-     * had crashed, unless a non-standard [CoroutineExceptionHandler] is installed in the context.
+     * In particular, it means that a parent coroutine invoking `join` on a child coroutine throws
```
Comment (qwwdfsad) [inline kotlinx-coroutines-core/common/src/Job.kt:255]: I'd suggest removing the whole paragraph, now it seems to be redundant. 
Also, please fix the same phrasing for `cancelAndJoin`

**Review by qwwdfsad — CHANGES_REQUESTED** [review]

File: kotlinx-coroutines-core/common/src/Job.kt (hunk 1 above)
Comment (elizarov) [reply kotlinx-coroutines-core/common/src/Job.kt:255]: I think this explanation still makes sense as it is quite counter-intuitive for people that you can crash an attempt to `join` another coroutine.

**Review by qwwdfsad — APPROVED** [review]

**Pull request merged**